src/sharding/ShardingManager.js

```diff
@@ -101,6 +109,54 @@ class ShardingManager extends EventEmitter {
     return shard;
   }
 
+  /**
+   * Kills all Shards without them respawning.
+   */
+  killAll() {
+    let chk = false;
+    if (this.respawn) {
+      this.respawn = false;
+      chk = true;
+    }
+    for (const shard of this.shards.values()) shard.kill();
+    if (chk) {
+      this.respawn = true;
+    }
+  }
+
+  /**
+   * Checks the Shard amount, then respawns the correct amount of Shards.
+   * @param {number} [delay=5500] How long to wait in between spawning each shard (in milliseconds)
+   * @param {boolean} [waitForReady=true] Whether to wait for a shard to become ready before continuing to another
+   * @returns {Promise<Collection<number, Shard>>}
+   */
+  async rebalanceShards(delay = 5500, waitForReady = true) {
```

Comment:
This methods looks like a bit of a combination of `killAll` and `spawn` to me.
You likely can cut the method quite a bit down by using those two methods.

I think something like this could do the trick:
1. Verify this.autoShards is true
2. Kill all shards
3. Clear all shards from collection to spawn correct shards later (see comment below)
4. Respawn with the spawn method, pass it `'auto'` to make it fetch the new count automatically

Reply:
Ive edited this Command, Please Check it over for me and Tell me if that is what your Looking for <3

Reply:
Ahh I did not pass it auto, my apologies. I did take your advice, and made the internal guts of the command, a balance between the functions.